### PR TITLE
Change get and request sticker string id parse methods to be more consistent

### PR DESCRIPTION
**Pull request by KILLEliteMaste**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](./CONTRIBUTING.md).

### Breaking Changes
- get/requestSticketById(String) methods do no longer throw an exception when the id is being parsed and instead return an empty optional or request a not existing sticker 

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/DiscordApi.java [hunk 1]
```diff
@@ -1436,7 +1436,7 @@ default CompletableFuture<Sticker> requestStickerById(String id) {
         try {
             return requestStickerById(Long.parseLong(id));
         } catch (NumberFormatException exception) {
-            throw new IllegalArgumentException("The ID must be a number!");
+            return requestStickerById(-1);
```
Comment (Saladoc) [inline javacord-api/src/main/java/org/javacord/api/DiscordApi.java:1439]:
```suggestion
            CompletableFuture<Sticker> future = new CompletableFuture<>();
            future.completeExceptionally(exception);
            return future;
```
There is no reason to swallow the exception and deprive the user of debugging information.

File: javacord-api/src/main/java/org/javacord/api/entity/server/Server.java [hunk 2]
```diff
@@ -3515,7 +3515,7 @@ default CompletableFuture<Sticker> requestStickerById(String id) {
         try {
             return requestStickerById(Long.parseLong(id));
         } catch (NumberFormatException exception) {
-            throw new IllegalArgumentException("The ID must be a number.");
+            return requestStickerById(-1);
```
Comment (Saladoc) [inline javacord-api/src/main/java/org/javacord/api/entity/server/Server.java:3518]:
```suggestion
            CompletableFuture<Sticker> future = new CompletableFuture<>();
            future.completeExceptionally(exception);
            return future;
```
Same as above. After the switch to Java 11 we can also do `return CompletableFuture.failedFuture(exception);`

**Review by Saladoc — CHANGES_REQUESTED** [review]

File: javacord-api/src/main/java/org/javacord/api/DiscordApi.java (hunk 1 above)
Comment (KILLEliteMaste) [reply javacord-api/src/main/java/org/javacord/api/DiscordApi.java:1439]: Seems there is another inconsistency. `DiscordApi#getUserById` looks the same like this

Comment (Saladoc) [reply javacord-api/src/main/java/org/javacord/api/DiscordApi.java:1439]: I'd argue we should return a failed future in both cases. Depending on how many methods have this problem we can create an issue or make the changes all in this PR. If we have a way to pass the exception to the user without breaking the method contract, we should do so and not throw away the information.

**Review by Saladoc — APPROVED** [review]

**Pull request merged**